File: bronze/1252/c++/main.cpp
#include <iostream>
#include <string>


int main()
{
    std::string t0, t1, answer = "", temp;
    std::cin >> t0 >> t1;
    int n0, n1, index = 0;
    if (t0.size() < t1.size()) {
        temp = t0;
        t0 = t1;
        t1 = temp;
    }
    n0 = t0.size() -1;
    n1 = t1.size() -1;
    int carry = 0, sum;
    while (n0 > -1) {
        if (n1 >= 0) {
            sum = (t0[n0] - '0') + (t1[n1] - '0') + carry;
            carry = sum / 2;
        } else if (n0 >= 0){
            sum = (t0[n0] - '0') + carry;
            carry = sum / 2;
        }
        
        answer += ('0' + (sum % 2));
        n0--;
        n1--;
    }
    if (carry) {
        answer += '1';
    }
    int i = answer.size() - 1;
    while (i > 0 && answer[i] == 0) {
        i--;
    }
    for (int i = answer.size() -1; i >= 0; i--) {
        std::cout << answer[i];
    }

    return 0;
}

File: bronze/1252/python/main.py

if (__name__ == "__main__"):
    t0, t1 = input().split()
    answer = ""
    if len(t0) < len(t1):
        t0, t1 = (t1, t0)
    
    n0 = len(t0) -1
    n1 = len(t1) -1

    carry = 0
    index = 0
    while n0 > -1:
        if n1 >= 0:
            sum = int(t0[n0]) + int(t1[n1]) + carry
            carry = sum // 2
        elif n0 >= 0:
            sum = int(t0[n0]) + carry
            carry = sum // 2
        answer += str(sum % 2)
        n0 -= 1
        n1 -= 1
    if carry: answer += '1'
    
    print(answer[::-1].lstrip('0') or '0')